packs/tests/actions/chains/test_quickstart_rules.yaml

```diff
@@ -28,7 +28,23 @@ chain:
         params:
             env:
               ST2_AUTH_TOKEN: "{{token}}"
-            cmd: "st2 rule create /usr/share/doc/st2/examples/rules/sample_rule_with_webhook.yaml &> /dev/null"
+            cmd: "st2 rule create /usr/share/doc/st2/examples/rules/sample_rule_with_webhook.yaml -j | grep name"
+        on-success: sleep
+    -
+        name: sleep
+        ref: core.local
+        params:
+            env:
+              ST2_AUTH_TOKEN: "{{token}}"
+            cmd: "sleep 5"
+        on-success: test_webhook_list    
+    -
+        name: test_webhook_list
+        ref: core.local
+        params:
+            env:
+              ST2_AUTH_TOKEN: "{{token}}"
+            cmd: "st2 webhook list -j | grep sample"
```

Comment:
You are probably right.  I'll look to moving that way during the next round
of refactoring.

PWH
On Jun 4, 2015 7:44 PM, "Tomaz Muraus" <notifications@github.com> wrote:

> In packs/tests/actions/chains/test_quickstart_rules.yaml
> <https://github.com/StackStorm/st2tests/pull/12#discussion_r31784254>:
>
> > +        on-success: sleep
> > +    -
> > +        name: sleep
> > +        ref: core.local
> > +        params:
> > +            env:
> > +              ST2_AUTH_TOKEN: "{{token}}"
> > +            cmd: "sleep 5"
> > +        on-success: test_webhook_list
> > +    -
> > +        name: test_webhook_list
> > +        ref: core.local
> > +        params:
> > +            env:
> > +              ST2_AUTH_TOKEN: "{{token}}"
> > +            cmd: "st2 webhook list -j | grep sample"
>
> It seems like that json query (http://stedolan.github.io/jq/) would come
> handy in the tests and make them more robust and fail safe.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StackStorm/st2tests/pull/12/files#r31784254>.
>